src/utils/debug.rs

```diff
@@ -0,0 +1,201 @@
+
+use console::Term;
+
+use std::ffi::OsString;
+use std::fs::File;
+use std::io::{Seek, SeekFrom, Write};
+use std::sync::atomic::AtomicBool;
+
+const RUST_BACKTRACE: &str = "RUST_BACKTRACE";
+
+// Use a global because the config where this might be stored could
+// itself panic while it is being built.
+static USING_DELTA_DEBUG_LOGFILE: AtomicBool = AtomicBool::new(false);
+
+#[derive(Debug)]
+pub struct PrintNoticeOnPanic;
+
+impl PrintNoticeOnPanic {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[cfg(not(test))]
+impl Drop for PrintNoticeOnPanic {
+    fn drop(&mut self) {
+        // Nothing elaborate with std::panic::set_hook or std::panic::catch_unwind to also get the backtrace,
+        // only set RUST_BACKTRACE when recording
+        if std::thread::panicking() {
+            if USING_DELTA_DEBUG_LOGFILE.load(DELTA_ATOMIC_ORDERING) {
+                if let Some(logfile) = std::env::var_os(DELTA_DEBUG_LOGFILE) {
+                    eprintln!("  Wrote {logfile:?} (if you want to share it, ensure no sensitive information is contained). You may also want to include the stack backtrace.");
+                }
+            } else {
+                // Setting GIT_PAGER however does not override interactive.diffFilter!
+                eprintln!("\n\
+            delta panicked and crashed, sorry :(\n\
+            To quickly repeat the previous command without delta, run 'export GIT_PAGER=less' first. If you want\n\
+            to report the crash and it is easy to repeat, do so after 'export DELTA_DEBUG_LOGFILE=crash.log'\n\
+            then submit the logfile to github at <https://github.com/dandavison/delta/issues>. Thank you!\n\n\
+            !! Make sure there is NO sensitive information in the log file, it will contain the entire diff !!\n\
+            ")
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+struct DeltaDetailInternal {
+    file: File,
+    bytes_written: u64,
+    max_log_size: u64,
+    truncate_back_to: u64,
+}
+
+pub struct RecordDeltaCall(Option<Box<DeltaDetailInternal>>);
+
+impl RecordDeltaCall {
+    pub fn new(config: &Config) -> Self {
+        fn make(logfile: &OsString, config: &Config) -> Result<Box<DeltaDetailInternal>, String> {
+            let mut file = File::create(logfile).map_err(|e| e.to_string())?;
+
+            let mut write = |s: String| file.write_all(s.as_bytes()).map_err(|e| e.to_string());
+
+            write(
+                "<details>\n\
+                <summary>Input which caused delta to crash</summary>\n```\n\n"
+                    .into(),
+            )?;
+
+            if let Some(cfg) = config.git_config.as_ref() {
+                write("git config values:\n".into())?;
+                cfg.for_each(".*", |entry, value: Option<&str>| {
+                    if !(entry.starts_with("user.")
+                        || entry.starts_with("remote.")
+                        || entry.starts_with("branch.")
+                        || entry.starts_with("gui."))
```

Comment:
Note: I inverted the logic, now printing all git configs (see `git config --get-regexp '.*'`) except a those listed here.